static/merge.js

```diff
@@ -400,4 +442,162 @@
+        const classId = $(this).attr('id').split('-')[2];
+        const construct = $(this).data('construct');
+        const constructId = construct.split('-')[1];
+
+        // Get <th> elements for students.
+        var students = $(`.star_chart_table-${constructId} tr`).find(`.class-${classId}`);
+
+        // Save cells to restore them when separating class.
+        const sublevels = saveCells(students, constructId, classId);
+
+        // Calculate amount of stars for each level inside merging class.
+        const star_amount = calculateStarAmountVerticalByLevel(sublevels, constructId, classId)
+
+        // Remove all empty <th> elements which starts row.
+        $(`.star_chart_table-${constructId}`).find(`.class-${classId}`).remove()
+
+        var tr = $(this).parent().parent();
+
+        // Append <th> with `className` to row which left.
+        $(tr).append(`<th scope="row" class="student-${constructId} class-${classId} thicker" ` +
+            `style="border-bottom: 3pt solid black; white-space: nowrap; ` +
+            `text-overflow: ellipsis; overflow: hidden; max-width: 150px;">${className}</th>`)
+
+        // Append to last row of class all stars for each level.
+        for (var i = 0; i < sublevels; i++) {
+            if (star_amount[i].length !== 0) {
+                let span = createSpan(star_amount[i]);
+
+                $(tr).append(`<td data-csl-id="" style="border-bottom: 3pt solid black;" ` +
+                    `class="text-center thicker" ` +
+                    `data-modal-launch-observations="[${star_amount[i]}]">` +
+                    `<i data-toggle="tooltip" title="" class="fa fa-star"></i>${span}</td>`)
+            } else {
+                $(tr).append('<td data-csl-id="" style="border-bottom: 3pt solid black;" ' +
+                    'class="text-center thicker"></td>')
+            }
+        }
+
+        $(this).hide();
+        $(`#class-separate-${classId}-${constructId}`).show();
+    })
+
+    $('.class-separate').click(function() {
+        const classId = $(this).attr('id').split('-')[2];
+        const construct = $(this).data('construct');
+        const constructId = construct.split('-')[1];
+
+        var tr = $(this).parent().parent();
+        var size = mergedVertical[constructId][classId].length;
+        var cells = $(tr).find('td');
+
+        // Get amount of sublevels used in table.
+        const sublevels = cells.length
+
+        // Remove all <td> elements inside created earlier row.
+        cells.remove();
+
+        // Remove header with class name
+        $(tr).find('th:eq(1)').remove();
+
+        // Append to row with button header with first student <th> element.
+        $(tr).append(mergedVertical[constructId][classId][size - 1].student);
+
+        // Append saved <td>s elements to row.
+        [...Array(sublevels).keys()].map(
+            i => $(tr).append(mergedVertical[constructId][classId][size - 1].cells[i]));
```

Comment:
is `i` a number? If not, please rename it to some more verbose name such as `key`. Take a note you have similar code below, so apply this change there as well.

Reply:
Yeah i is a number. `[...Array(sublevels).keys()]` this creates for me array with range from 0 to sublevels value.